### ISSUE TITLE
Terraform Override Support

### ISSUE THREAD

**Issue by jhop310** [bridgecrewio/checkov#1779]
**Describe the bug**
Currently if you provide a terraform override file, as defined [here](https://www.terraform.io/docs/language/files/override.html) checkov will provide a false positive on the override resource because it is not a full resource definition pre-merge.

**To Reproduce**

Example compliant s3 bucket resource in main.tf:

``` terraform
resource "aws_s3_bucket" "example" {
  bucket        = local.s3_name
  acl           = "log-delivery-write"
  force_destroy = false

  server_side_encryption_configuration {
    rule {
      apply_server_side_encryption_by_default {
        sse_algorithm = "AES256"
      }
    }
  }

  versioning {
    enabled = true
  }

  lifecycle {
    prevent_destroy = true
  }
}
```

Example override.tf
``` terraform 
resource "aws_s3_bucket" "example" {
  force_destroy = true
  lifecycle {
    prevent_destroy = false
  }
}
```

Checkov output:
``` shell
Resource aws_s3_bucket.example failed in check CKV_AWS_19/Ensure all data stored in the S3 bucket is securely encrypted at rest - .../override/destroy_override.tf:[1, 6] - Guideline: https://docs.bridgecrew.io/docs/s3_14-data-encrypted-at-rest
```
**Expected behavior**

Checkov should either recognize and determine what the merged resource definition would look like, and scan that-  Or ignore terraform overrides all together.
